scripts/lsp_types.lua

```diff
@@ -18,19 +19,82 @@ local function tofile(fname, text)
   end
 end
 
-function M.gen(opt)
-  if vim.uv.fs_stat('./lsp.json') then
-    vim.fn.delete('./lsp.json')
+local function read_json(opt)
+  local uri = 'https://raw.githubusercontent.com/microsoft/language-server-protocol/gh-pages/_specifications/lsp/'
+    .. opt.version
+    .. '/metaModel/metaModel.json'
+
+  local res = vim.system({ 'curl', uri, '-o', '-' }):wait()
+
+  if res.code ~= 0 then
+    io.write(res.stderr)
+    return
+  end
+  return vim.json.decode(res.stdout)
+end
+
+local function gen_methods(protocol)
+  local output =
+    { '--- LSP methods. Generated by lsp_types.lua, keep at end of file.', 'protocol.Methods = {' }
+  local indent = (' '):rep(2)
+
+  for _, item in ipairs(protocol.requests) do
```

Comment:
Did you try `vim.spairs` ? So the generated `Methods` fields will be sorted.

Reply:
before tried it, it works for list-lie tables. but the result same as `ipairs` ..you want sort by `textDoucment`?

Reply:
ok, did you try sorting `protocol.requests` with `table.sort` ? The goal is the sort the result somehow.

Reply:
hmmm the doc said it works for list-like
```
    Parameters:  
      • {t}  (table) List-like table
```
and what is the meaning of sort in there every item struct in `protocol.request` like

```
{
  documentation = "A request send from the client to the server to execute a command. The request might return\na workspace edit which the client will apply to the workspace.",
  messageDirection = "clientToServer",
  method = "workspace/executeCommand",
  params = {
    kind = "reference",
    name = "ExecuteCommandParams"
  },
  registrationOptions = {
    kind = "reference",
    name = "ExecuteCommandRegistrationOptions"
  },
  result = {
    items = { {
        kind = "reference",
        name = "LSPAny"
      }, {
        kind = "base",
        name = "null"
      } },
    kind = "or"
  }
}
```

Reply:
> and what is the meaning of sort in there every item struct in `protocol.request` like

sort by the `method` field

Reply:
right. so I tried use `find:(textDocument)` before. but some method does not have prefix or have other prefix like `$` if sort they should in first. so I removed it .if you want this I will do it again.

Reply:
extract the name cleanup logic (which is needed below, anyway) to a function and use that in the sort

Reply:
how about now ?